packages/replay-internal/src/util/logger.ts

```diff
@@ -0,0 +1,111 @@
+import { addBreadcrumb, captureException } from '@sentry/core';
+import type { ConsoleLevel, SeverityLevel } from '@sentry/types';
+import { logger as coreLogger } from '@sentry/utils';
+
+import { DEBUG_BUILD } from '../debug-build';
+
+type ReplayConsoleLevels = Extract<ConsoleLevel, 'info' | 'warn' | 'error' | 'log'>;
+const CONSOLE_LEVELS: readonly ReplayConsoleLevels[] = ['info', 'warn', 'error', 'log'] as const;
+const PREFIX = '[Replay] ';
+
+type LoggerMethod = (...args: unknown[]) => void;
+type LoggerConsoleMethods = Record<ReplayConsoleLevels, LoggerMethod>;
+
+interface LoggerConfig {
+  captureExceptions: boolean;
+  traceInternals: boolean;
+}
+
+/** JSDoc */
+interface ReplayLogger extends LoggerConsoleMethods {
+  /**
+   * Calls `logger.info` but saves breadcrumb in the next tick due to race
+   * conditions before replay is initialized.
+   */
+  infoTick: LoggerMethod;
+  /**
+   * Captures exceptions (`Error`) if "capture internal exceptions" is enabled
+   */
+  exception: LoggerMethod;
+  /**
+   * Configures the logger with additional debugging behavior
+   */
+  setConfig(config: LoggerConfig): void;
+}
+
+function _addBreadcrumb(message: unknown, level: SeverityLevel = 'info'): void {
+  // Only support strings for breadcrumbs
+  if (typeof message !== 'string') {
```

Comment:
l: Do we need this check? IMHO we generally use this ourselves, worst case ``${message}`` will convert whatever it is to a string anyhow, which I guess is OK here...?